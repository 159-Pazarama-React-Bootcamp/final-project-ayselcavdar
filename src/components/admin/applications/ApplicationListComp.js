import React,{ useState } from 'react';
import styles from './ApplicationListComp.module.css';
import ApplicationCard from '../../applications/ApplicationCard';
import Button from '../../button/Button';
import { useDispatch, useSelector } from 'react-redux';
import { logoutInitiate } from '../../../redux/actions/logoutActions';
import Dropdown from '../../dropdown/Dropdown';
import { options, statusEnum } from '../../../constants/enum';
import Loader from '../../loader/Loader';
import Alert from '../../alert/Alert';

const ApplicationListComp = () => {
  const [values, setValues] = useState({
    queryCode: '',
    filterVal: '',
  });
  const { user, data } = useSelector((state) => state);
  const dispatch = useDispatch();

  const handleChange = (e) => {
    const { name, value } = e.target;
    setValues({ ...values, [name]: value });
  };

  const handleLogout = () => {
    user.currUser && dispatch(logoutInitiate());
  };

  // default admin cevaplanmamış başvuruları görüntülemekte
  // isterse cevaplanmış başvuruları da filtreleyerek görüntüleyebilmektedir.
  const applicationList = data?.applications
    ?.filter((item) => {
      if (values.filterVal !== '' && item.status !== values.filterVal)
        return false;
      if (!item.id.toLowerCase().includes(values.queryCode.toLowerCase()))
        return false;
      return true;
    })
    .filter(
      (item) => values.filterVal == statusEnum.ANSWERED.name || item.status == statusEnum.EVALUATING.name
    )
    .map((app) => <ApplicationCard app={app} key={app.id} to={`${app.id}`} />);

  return (
    <div className={styles['main-container']}>
      <div className={styles['header-container']}>
        <div className={styles['title-admin']}>
          <h2>Cevap bekleyen başvurular</h2>
          <Button onClick={handleLogout} type="reset" content={'çıkış'} />
        </div>
        <div className={styles['filter-container']}>
          <div>
            <input
              type={'text'}
              placeholder="Başvuru kodu ile filtreleyiniz."
              name="queryCode"
              value={values.queryCode}
              onChange={handleChange}
              className={styles['filter-input--admin']}
            />
          </div>
          <Dropdown
            label="Başvuru durumuna göre filtereleyiniz"
            name="filterVal"
            data={options}
            onChange={handleChange}
            value={values.filterVal}
            customClass={styles['custom-dropdown']}
          />
        </div>
      </div>
      <div className={styles['card-container']}>
        {data?.loading ? (
          <div className={styles['animation-container']}>
            <Loader />
          </div>
        ) : applicationList.length === 0 && !data?.error ? (
          <Alert
            title="Bilgilendirme"
            info
            body={'Herhangi bir başvuru bulunamadı.'}
          />
        ) : data?.error ? (
          <Alert title="Hata" error body={`${data?.error}`} />
        ) : (
          applicationList
        )}
      </div>
    </div>
  );
};

export default ApplicationListComp;
