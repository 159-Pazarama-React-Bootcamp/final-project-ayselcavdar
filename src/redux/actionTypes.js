const LOGIN_START = 'LOGIN_START';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_FAIL = 'LOGIN_FAIL';

const LOGOUT_START = 'LOGOUT_START';
const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
const LOGOUT_FAIL = 'LOGOUT_FAIL';

const SET_USER = 'SET_USER';

const GET_APPLICATIONS_START = 'GET_APPLICATIONS_START';
const GET_APPLICATIONS_SUCCESS = 'GET_APPLICATIONS_SUCCESS';
const GET_APPLICATIONS_FAIL = 'GET_APPLICATIONS_FAIL';

const DELETE_APPLICATION_START = 'DELETE_APPLICATION_START';
const DELETE_APPLICATION_FAIL = 'DELETE_APPLICATION_FAIL';
const DELETE_APPLICATION_SUCCESS = 'DELETE_APPLICATION_SUCCESS';

const CREATE_APPLICATION_START = 'CREATE_APPLICATION_START';
const CREATE_APPLICATION_SUCCESS = 'CREATE_APPLICATION_SUCCESS';
const CREATE_APPLICATION_FAIL = 'CREATE_APPLICATION_FAIL';

const EDIT_APPLICATION_START = 'EDIT_APPLICATION_START';
const EDIT_APPLICATION_FAIL = 'EDIT_APPLICATION_FAIL';
const EDIT_APPLICATION_SUCCESS = 'EDIT_APPLICATION_SUCCESS';


export {
  LOGIN_START,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_START,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL,
  SET_USER,
  CREATE_APPLICATION_FAIL,
  CREATE_APPLICATION_START,
  CREATE_APPLICATION_SUCCESS,
  GET_APPLICATIONS_FAIL,
  GET_APPLICATIONS_START,
  GET_APPLICATIONS_SUCCESS,
  DELETE_APPLICATION_START,
  DELETE_APPLICATION_SUCCESS,
  DELETE_APPLICATION_FAIL,
  EDIT_APPLICATION_FAIL,
  EDIT_APPLICATION_START,
  EDIT_APPLICATION_SUCCESS,
};
