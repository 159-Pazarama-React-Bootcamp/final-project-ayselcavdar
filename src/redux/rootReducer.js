import {combineReducers} from 'redux';
import { authReducer } from './reducers/authReducer';
import applicationReducer from './reducers/applicationReducer';

const reducers = {
  user: authReducer,
  data: applicationReducer,
};

export const rootReducer = combineReducers(reducers);