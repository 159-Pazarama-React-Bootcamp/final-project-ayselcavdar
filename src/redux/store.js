import {createStore, applyMiddleware} from 'redux';
import {rootReducer} from './rootReducer';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middleware = [thunk];
// state değişikliklerini takibe yarayan anlık loglar sunan logger modulü,
// yalnızca development ortamı için uygulanmıştır
if(process.env.NODE_ENV === 'development'){
    middleware.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middleware));