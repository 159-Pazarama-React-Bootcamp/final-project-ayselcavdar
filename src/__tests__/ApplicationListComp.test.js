/**
 * @jest-environment jsdom
 */
import render from 'react-test-renderer';
import ApplicationListComp from '../components/admin/applications/ApplicationListComp';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

describe('<ApplicationListComp /> rendering', () => {
  test('AppDetails snapshots check', () => {
    const wrapper = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <ApplicationListComp />
        </BrowserRouter>
      </Provider>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
