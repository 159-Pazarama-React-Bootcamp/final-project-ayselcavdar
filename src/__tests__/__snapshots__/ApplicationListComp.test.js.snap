// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationListComp /> rendering AppDetails render check 1`] = `
<div
  className="main-container"
>
  <div
    className="header-container"
  >
    <div
      className="title-admin"
    >
      <h2>
        Cevap bekleyen başvurular
      </h2>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "color": "#6B63FF",
          }
        }
        type="reset"
      >
        çıkış
      </button>
    </div>
    <div
      className="filter-container"
    >
      <div>
        <input
          className="filter-input--admin"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodu ile filtreleyiniz."
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group--select custom-dropdown"
      >
        <select
          className="form-control"
          name="filterVal"
          onChange={[Function]}
          placeholder=""
          value=""
        >
          <option
            value=""
          >
            Başvuru durumuna göre filtereleyiniz
          </option>
          <option
            value="EVALUATING"
          >
            EVALUATING
          </option>
          <option
            value="ANSWERED"
          >
            ANSWERED
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="alert-container alert-info "
    >
      <div
        className="title-container"
      >
        <h3
          className="title-text"
        >
          Bilgilendirme
        </h3>
      </div>
      <div
        className="body-container"
      >
        <p
          className="body-text"
        >
          Herhangi bir başvuru bulunamadı.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ApplicationListComp /> rendering AppDetails snapshots check 1`] = `
<div
  className="main-container"
>
  <div
    className="header-container"
  >
    <div
      className="title-admin"
    >
      <h2>
        Cevap bekleyen başvurular
      </h2>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "color": "#6B63FF",
          }
        }
        type="reset"
      >
        çıkış
      </button>
    </div>
    <div
      className="filter-container"
    >
      <div>
        <input
          className="filter-input--admin"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodu ile filtreleyiniz."
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group--select custom-dropdown"
      >
        <select
          className="form-control"
          name="filterVal"
          onChange={[Function]}
          placeholder=""
          value=""
        >
          <option
            value=""
          >
            Başvuru durumuna göre filtereleyiniz
          </option>
          <option
            value="EVALUATING"
          >
            EVALUATING
          </option>
          <option
            value="ANSWERED"
          >
            ANSWERED
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="alert-container alert-info "
    >
      <div
        className="title-container"
      >
        <h3
          className="title-text"
        >
          Bilgilendirme
        </h3>
      </div>
      <div
        className="body-container"
      >
        <p
          className="body-text"
        >
          Herhangi bir başvuru bulunamadı.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppDetails render check 1`] = `
<div
  className="main-container"
>
  <div
    className="header-container"
  >
    <div
      className="title-admin"
    >
      <h2>
        Cevap bekleyen başvurular
      </h2>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "color": "#6B63FF",
          }
        }
        type="reset"
      >
        çıkış
      </button>
    </div>
    <div
      className="filter-container"
    >
      <div>
        <input
          className="filter-input--admin"
          name="queryCode"
          onChange={[Function]}
          placeholder="Başvuru kodu ile filtreleyiniz."
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group--select custom-dropdown"
      >
        <select
          className="form-control"
          name="filterVal"
          onChange={[Function]}
          placeholder=""
          value=""
        >
          <option
            value=""
          >
            Başvuru durumuna göre filtereleyiniz
          </option>
          <option
            value="EVALUATING"
          >
            EVALUATING
          </option>
          <option
            value="ANSWERED"
          >
            ANSWERED
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="card-container"
  >
    <div
      className="alert-container alert-info "
    >
      <div
        className="title-container"
      >
        <h3
          className="title-text"
        >
          Bilgilendirme
        </h3>
      </div>
      <div
        className="body-container"
      >
        <p
          className="body-text"
        >
          Herhangi bir başvuru bulunamadı.
        </p>
      </div>
    </div>
  </div>
</div>
`;
